code/go/0chain.net/smartcontract/zcnsc/mint.go

```diff
@@ -86,11 +87,20 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
 		return
 	}
 
+	numAuth, err := getAuthorizerCount(ctx)
+
 	// verify signatures of authorizers
 	err = payload.verifySignatures(ctx)
 	if err != nil {
 		msg := fmt.Sprintf("failed to verify signatures with error: %v, %s", err, info)
 		err = common.NewError(code, msg)
+	}
+
+	if payload.countUniqueSignatures() < int(math.RoundToEven(gn.PercentAuthorizers*float64(numAuth))) {
```

Comment:
Unique signatures checking should be done in `payload.verifySignatures` to avoid duplicate verification. The signature verify process is cpu immense, so if any of the signatures are duplicated, we should detect early.

Reply:
hey @peterlimg, do we need to delete duplicate signatures from `payload.Signatures`?